coconut/src/proof/mod.rs

```diff
@@ -178,7 +178,7 @@ impl<'pair, Pair, F: PrimeField> UnpackedBlindedMessages<'pair, Pair, F> {
         let (paired, (msgs, blindings)): (Vec<_>, _) =
             process_results(paired, |iter| iter.unzip())?;
         if paired.is_empty() {
```

Comment:
You can remove this `if` block as the last `else` block will return the desired result.